packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -47,6 +47,24 @@ export function truncate(fileString: string, truncateMarker: RegExp): string {
   return fileString.split(truncateMarker, 1).shift()!;
 }
 
+function reportTruncateMarkerProblem({
+  blogSourceRelative,
+  truncateMarker,
+  options,
+  content,
+}: {
+  content: string;
+  truncateMarker: RegExp;
+  blogSourceRelative: string;
+  options: Pick<PluginOptions, 'onUntruncatedBlogPost'>;
+}): void {
+  if (!truncateMarker.test(content)) {
+    logger.report(options.onUntruncatedBlogPost)(
+      logger.interpolate`Blog post path=${blogSourceRelative} is not truncated.`,
+    );
+  }
+}
```

Comment:
We shouldn't run this Regex against the content twice: this can be expensive and is useless work.

We already run it later in the blog post processing to compute the `hasTruncateMarker` meta attribute.

I'd prefer if:
- we emit the warning based on `blogPost.metadata.hasTruncateMarker`
- we emit a single report for all the blog posts
- we explain in the message that this can be turned off with the blog `onUntruncatedBlogPost: 'ignore'` option. Many sites run with untruncated blogs on purpose, let's give them a better upgrading experience if warn is the default.


The function should rather be something like this:

```ts
function reportTruncateMarkerProblem({
  blogPosts,
  onUntruncatedBlogPost,
}: {
  blogPosts: BlogPost[],
  onUntruncatedBlogPost: PluginOptions['onUntruncatedBlogPost']
}): void {
  const untruncatedBlogPosts = blogPosts.filter(p => !p.metadata.hasTruncateMarker)
  if (onUntruncatedBlogPost !== "ignore" && onUntruncatedBlogPost.length > 0) {
    const message = `Blabla there are untruncated blog posts:
${untruncatedBlogPosts.map(getPath).join('\n- ')}
Blabla additional help`
    logger.report(options.onUntruncatedBlogPost)(message);
  }
}
```